backend/functions/lib/adapters/gateways/dxtr/offices.dxtr.gateway.test.ts

```diff
@@ -0,0 +1,141 @@
+      applicationContext = await createMockApplicationContext({ DATABASE_MOCK: 'true' });
+      querySpy.mockImplementation(jest.fn());
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    test('Should get Offices', async () => {
+      const mockResults: QueryResults = {
+        success: true,
+        results: {
+          recordset: OFFICES,
+        },
+        message: '',
+      };
+      querySpy.mockImplementation(async () => {
+        return Promise.resolve(mockResults);
+      });
+
+      const gateway = new OfficesDxtrGateway();
+
+      const offices = await gateway.getOffices(applicationContext);
+      expect(offices).toEqual(OFFICES);
+    });
+
+    test('should throw error when success is false', async () => {
+      const mockResults: QueryResults = {
+        success: false,
+        results: {},
+        message: 'Some expected SQL error.',
+      };
+      querySpy.mockImplementation(async () => {
+        return Promise.resolve(mockResults);
+      });
+
+      const gateway = new OfficesDxtrGateway();
+
+      await expect(async () => {
+        await gateway.getOffices(applicationContext);
+      }).rejects.toThrow('Some expected SQL error.');
+    });
+  });
+
+  describe('getOfficeByCourtIdAndOfficeCode test', () => {
+    let applicationContext: ApplicationContext;
+    const querySpy = jest.spyOn(database, 'executeQuery');
+
+    beforeEach(async () => {
+      applicationContext = await createMockApplicationContext({ DATABASE_MOCK: 'true' });
+      querySpy.mockImplementation(jest.fn());
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    test('Should get office by office code and courtid', async () => {
+      const mockResults: QueryResults = {
+        success: true,
+        results: {
+          recordset: [MANHATTAN],
+        },
+        message: '',
+      };
+      querySpy.mockImplementation(async () => {
+        return Promise.resolve(mockResults);
+      });
+
+      const gateway = new OfficesDxtrGateway();
+
+      const offices = await gateway.getOfficeByCourtIdAndOfficeCode(
+        applicationContext,
+        '0081',
+        '1',
+      );
+      expect(offices).toEqual(MANHATTAN);
+    });
+
+    test('should throw error when success is false', async () => {
+      const gateway = new OfficesDxtrGateway();
+      await expect(async () => {
+        await gateway.getOfficeByCourtIdAndOfficeCode(applicationContext, '081', '1');
+      }).rejects.toThrow('Invalid court id or office code supplied');
+
+      await expect(async () => {
+        await gateway.getOfficeByCourtIdAndOfficeCode(applicationContext, '0081', '01');
+      }).rejects.toThrow('Invalid court id or office code supplied');
+    });
+
+    test('should throw CamsError when success is false', async () => {
+      const mockResults: QueryResults = {
+        success: false,
+        results: {},
+        message: 'Some expected SQL error.',
+      };
```

Comment:
I know these 2 tests are different but the description is almost the same.  A better description would be helpful.